src/metrics/gossip_metrics.go

```diff
@@ -239,24 +249,30 @@ func (c *GossipMetrics) FillMetrics(ep track.ExtendedPeerstore) {
 					peerMetrics.Ip = ip
 					peerMetrics.Country = country
 					peerMetrics.City = city
+					newMetaFlag = true
+					msg.Country = country
+					msg.City = city
 				}
 			}
-
 			// Since we want to have the latest Latency, we update it only when it is different from 0
 			// latency in seconds
 			if peerData.Latency != 0 {
 				peerMetrics.Latency = float64(peerData.Latency/time.Millisecond) / 1000
+				newMetaFlag = true
+				msg.Latency = fmt.Sprintf("%f", float64(peerData.Latency/time.Millisecond)/1000)
 			}
 
+			// Check if there is new metadata to report to the Armiarma Server
+			if newMetaFlag {
+				fmt.Println("New Msg to send")
+				// add the metadata fields to the msg
+				// msg.MetadataRequest = peerData.MetadataRequest
+				// msg.MetadataSucceed = peerData.MetadataSucced
+				// notify of new message to send
+				c.ArmiarmaServer.NewPeerMetadata <- msg
```

Comment:
Related to one of my other comments:

- I would directly call the function without relying on channels.
- If both "new peer discovered" and "new peer metadata detected" share the same rpc call, we can share the same code.